src/js/actions/tools.js

```diff
@@ -797,8 +795,28 @@ define(function (require, exports) {
      */
     var beforeStartup = function () {
         var flux = this.flux,
+            appStore = flux.store("application"),
             toolStore = this.flux.store("tool"),
-            tools = toolStore.getAllTools();
+            tools = toolStore.getAllTools(),
+            documentLayerBounds;
+
+        var throttledResetBorderPolicies =
+            synchronization.throttle(this.flux.actions.tools.resetBorderPolicies, this, 100),
+            _documentOrApplicationChangeHandler = function () {
+                var currentDocument = appStore.getCurrentDocument();
+
+                if (currentDocument) {
```

Comment:
Seems like you want to clean up the `documentLayerBounds` if there is no current document?